container-host-files/etc/hcf/config/scripts/configure-HA-hosts.sh

```diff
@@ -10,60 +10,62 @@
 # its target, which the low-level network setup of the system
 # round-robins to all the machines in the etcd sub-cluster.
 
-find_cluster_ha_hosts() {
-    local component_name="${1}"
-    if test -z "${KUBERNETES_SERVICE_HOST:-}" ; then
-        # on Vagrant ; HA is not supported
-        # Fall back to simple hostname
-        echo "[\"${component_name}\"]"
-        return 0
-    fi
+set -x
 
-    local this_component="${HOSTNAME%-*}"
-    if test "${this_component}" != "${component_name}" ; then
-        echo "[\"${component_name}\"]"
-        return 0
-    fi
+k8s_api() {
+    local svcacct=/var/run/secrets/kubernetes.io/serviceaccount
+    curl --silent \
+        --cacert "${svcacct}/ca.crt" \
+        -H "Authorization: bearer $(cat "${svcacct}/token")" \
+        "https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/api/v1/namespaces/$(cat "${svcacct}/namespace")/${1#/}"
+}
 
-    if test -n "${KUBERNETES_NAMESPACE:-}" ; then
-        # This is running on raw Kubernetes
+find_cluster_ha_hosts() {
+    local component_name this_component hosts
+    component_name="${1}"
+    this_component="$(k8s_api "/pods/${HOSTNAME}" | jq -crM '.metadata.labels."skiff-role-name"')"
 
-        # Loop over the environment to locate the component name variables.
-        local hosts=''
-        local names=()
+    if test "${this_component}" != "${component_name}" ; then
+        # Requesting a different component, use DNS name
+        echo "[\"${component_name}.${K8S_SERVICE_DOMAIN_SUFFIX}\"]"
+    else
+        # Requesting all the pods in this component
         local i
         for (( i = 0 ; i < 60 ; i ++ )) ; do
-            names=($(dig "${component_name}.${HCP_SERVICE_DOMAIN_SUFFIX}" -t SRV | awk '/IN[\t ]+A/ { print $1 }'))
-            if test "${#names[@]}" -gt 0 ; then
+            hosts="$(k8s_api /pods | jq -crM '
+                .items |
+                map(
+                    select(.metadata.labels."skiff-role-name" == "'"${component_name}"'") |
+                    .metadata.name |
+                    select(.) |
+                    . + "'".${component_name}-pod.${K8S_SERVICE_DOMAIN_SUFFIX}"'"
```

Comment:
I think we should rename the `-pod` convention to `-set` here: https://github.com/SUSE/fissile/blob/master/kube/stateful_set.go#L47